chainio/txmgr/geometric/geometric.go

```diff
@@ -0,0 +1,598 @@
+	gasLimit, err := t.ethClient.EstimateGas(ctx, ethereum.CallMsg{
+		From:      from,
+		To:        tx.To(),
+		GasTipCap: newGasTipCap,
+		GasFeeCap: gasFeeCap,
+		Value:     tx.Value(),
+		Data:      tx.Data(),
+	})
+	if err != nil {
+		return nil, utils.WrapError("failed to estimate gas", err)
+	}
+	// we also add a buffer to the gas limit to account for potential changes in the state of the chain
+	// between the time of estimation and the time the transaction is mined
+	bufferedGasLimit := t.addGasBuffer(gasLimit)
+
+	return types.NewTx(&types.DynamicFeeTx{
+		ChainID:    tx.ChainId(),
+		Nonce:      tx.Nonce(),
+		GasTipCap:  newGasTipCap,
+		GasFeeCap:  gasFeeCap,
+		Gas:        bufferedGasLimit,
+		To:         tx.To(),
+		Value:      tx.Value(),
+		Data:       tx.Data(),
+		AccessList: tx.AccessList(),
+	}), nil
+}
+
+func (t *GeometricTxManager) estimateGasTipCap(ctx context.Context) (gasTipCap *big.Int, err error) {
+	gasTipCap, err = t.ethClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		// If the transaction failed because the backend does not support
+		// eth_maxPriorityFeePerGas, fallback to using the default constant.
+		// Currently Alchemy is the only backend provider that exposes this
+		// method, so in the event their API is unreachable we can fallback to a
+		// degraded mode of operation. This also applies to our test
+		// environments, as hardhat doesn't support the query either.
+		// TODO: error could actually come from node not being down or network being slow, etc.
+		t.logger.Info("eth_maxPriorityFeePerGas is unsupported by current backend, using fallback gasTipCap")
+		gasTipCap = big.NewInt(0).SetUint64(t.params.FallbackGasTipCap)
+	}
+	return t.addGasTipCapBuffer(gasTipCap), nil
+}
+
+// addGasTipCapBuffer adds a buffer to the gas tip cap to account for potential changes in the state of the chain
+// The result is returned in a new big.Int to avoid modifying the input gasTipCap.
+func (t *GeometricTxManager) addGasTipCapBuffer(gasTipCap *big.Int) *big.Int {
+	bumpedGasTipCap := new(big.Int).Set(gasTipCap)
+	return bumpedGasTipCap.Mul(bumpedGasTipCap, big.NewInt(int64(t.params.GasTipMultiplierPercentage))).
+		Div(bumpedGasTipCap, big.NewInt(100))
+}
+
+// estimateGasFeeCap returns the gas fee cap for a transaction, calculated as:
+// gasFeeCap = 2 * baseFee + gasTipCap
+// Rationale: https://www.blocknative.com/blog/eip-1559-fees
+// The result is returned in a new big.Int to avoid modifying gasTipCap.
+func (t *GeometricTxManager) estimateGasFeeCap(ctx context.Context, gasTipCap *big.Int) (*big.Int, error) {
+	header, err := t.ethClient.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return nil, utils.WrapError("failed to get latest header", err)
+	}
+	return new(big.Int).Add(new(big.Int).Mul(header.BaseFee, big.NewInt(2)), gasTipCap), nil
+}
+
+func (t *GeometricTxManager) addGasBuffer(gasLimit uint64) uint64 {
+	return t.params.GasMultiplierPercentage * gasLimit / 100
```

Comment:
why do we keep GasMultiplier in %age and then divide by 100 instead of just doing it as multiplier. 1.2 instead of 120?

Reply:
same ques for other multipliers

Reply:
converting to float and back, like `return uint64(t.params.GasMultiplierFloat * float64(gasLimit))` could work, but if the gasLimit is big number (not sure what the cutoff is), there could be floating point rounding errors. chatGPT seems to suggest for float64 numbers, the range 10^16 start having gaps between integers. gasLimit will always be below that (30M/block), so we could potentially use the floating point approach. Do you think that would make the param more easy to understand (setting 1.2 instead of 120)?

Reply:
yea I think `gasMultiplierPercentage` seems overloaded. To me `gasMultiplier` seems simpler to understand.

Reply:
[91d36d1](https://github.com/Layr-Labs/eigensdk-go/pull/284/commits/91d36d1b0df6569da58cb6edd65c89ff296ae644)